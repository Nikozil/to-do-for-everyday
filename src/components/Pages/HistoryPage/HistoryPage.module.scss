@import '../../../App.scss';
.wrapper {
  height: 100%;
}
.days-list {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));
  justify-content: space-between;
  grid-gap: 5px;
  padding: 0 15px;
}
.spinner-wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 50px;
}

.spinner {
  color: $primary-color !important;
}
