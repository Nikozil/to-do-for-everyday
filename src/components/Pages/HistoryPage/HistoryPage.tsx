import React from 'react';
import { useSelector } from 'react-redux';

import { AppStateType } from '../../../Redux/store';
import SpinComponent from '../../../assets/SpinComponent/SpinComponent';
import DayComponent from './DayComponent/DayComponent';
import MonthPaginator from './MonthPaginator/MonthPaginator';
import styles from './HistoryPage.module.scss';

const HistoryPage = () => {
  const days = useSelector((state: AppStateType) => state.history.daysList);
  const initHistoryStatus = useSelector(
    (state: AppStateType) => state.history.initStatus
  );

  return (
    <div className={styles['history-page']}>
      <MonthPaginator />
      {initHistoryStatus ? (
        <ul className={styles['days-list']}>
          {days.length
            ? days.map((day) => <DayComponent key={day.timestamp} day={day} />)
            : null}
        </ul>
      ) : (
        <div className={styles['preloader']}>
          <SpinComponent styleClass={styles['preloader__spinner']} />
        </div>
      )}
    </div>
  );
};

export default HistoryPage;
