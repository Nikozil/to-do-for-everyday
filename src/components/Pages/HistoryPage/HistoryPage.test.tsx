import React from 'react';
import { render } from '../../../utils/tests/test-utils';
import HistoryPage from './HistoryPage';

describe('HistoryPage', () => {
  it('render correct', async () => {
    const { asFragment } = render(<HistoryPage />, {
      preloadedState: {
        // history: {
        //   ...initialState,
        //   days: [
        //     {
        //       timestamp: 1635263950643,
        //       score: 3,
        //       tag: 'hi',
        //       doneTasksList: [{ name: 'do', id: '123' }],
        //     },
        //   ],
        //   initStatus: true,
        // },
        clock: { time: 1635263951643 },
      },
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
