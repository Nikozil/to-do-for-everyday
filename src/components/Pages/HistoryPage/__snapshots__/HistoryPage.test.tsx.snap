// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryPage render correct 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="wrapper"
    >
      <nav
        aria-label="paginator "
        class="nav"
      >
        <ul
          class="pagination pagination-sm flex-wrap justify-content-center mb-1"
        >
          <li
            class="page-item button"
          >
            <span
              class="page-link name"
            >
              2020
            </span>
          </li>
          <li
            class="page-item button active"
          >
            <span
              class="page-link name active"
            >
              2021
            </span>
          </li>
        </ul>
      </nav>
      <nav
        aria-label="paginator "
        class="nav"
      >
        <ul
          class="pagination pagination-sm flex-wrap justify-content-center mb-1"
        >
          <li
            class="page-item button"
          >
            <span
              class="page-link name"
            >
              1
            </span>
          </li>
          <li
            class="page-item button"
          >
            <span
              class="page-link name"
            >
              2
            </span>
          </li>
          <li
            class="page-item button"
          >
            <span
              class="page-link name"
            >
              3
            </span>
          </li>
          <li
            class="page-item button"
          >
            <span
              class="page-link name"
            >
              4
            </span>
          </li>
          <li
            class="page-item button"
          >
            <span
              class="page-link name"
            >
              5
            </span>
          </li>
          <li
            class="page-item button"
          >
            <span
              class="page-link name"
            >
              6
            </span>
          </li>
          <li
            class="page-item button"
          >
            <span
              class="page-link name"
            >
              7
            </span>
          </li>
          <li
            class="page-item button"
          >
            <span
              class="page-link name"
            >
              8
            </span>
          </li>
          <li
            class="page-item button"
          >
            <span
              class="page-link name"
            >
              9
            </span>
          </li>
          <li
            class="page-item button active"
          >
            <span
              class="page-link name active"
            >
              10
            </span>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="spinner-wrapper"
    >
      <div
        class="spinner-border spinner"
        role="status"
      >
        <span
          class="visually-hidden"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
   
</DocumentFragment>
`;
